calyx-opt/src/passes/compile_static.rs

```diff
@@ -239,6 +239,81 @@ impl CompileStatic {
             early_reset_group.borrow().attributes.clone();
         g
     }
+
+    fn get_reset_group_name(&self, sc: &mut ir::StaticControl) -> &ir::Id {
+        // assume that there are only static enables left.
+        // if there are any other type of static control, then error out.
+        let ir::StaticControl::Enable(s) = sc else {
+            unreachable!("Non-Enable Static Control should have been compiled away. Run {} to do this", crate::passes::StaticInliner::name());
+        };
+
+        let sgroup = s.group.borrow_mut();
+        let sgroup_name = sgroup.name();
+        // get the "early reset group". It should exist, since we made an
+        // early_reset group for every static group in the component
+        let early_reset_name =
+            self.reset_early_map.get(&sgroup_name).unwrap_or_else(|| {
+                unreachable!(
+                    "group {} not in self.reset_early_map",
+                    sgroup_name
+                )
+            });
+
+        early_reset_name
+    }
+
+    fn build_wrapper_group_while(
+        &self,
+        fsm_name: &ir::Id,
+        fsm_width: u64,
+        group_name: &ir::Id,
+        port: RRC<ir::Port>,
+        builder: &mut ir::Builder,
+    ) -> RRC<ir::Group> {
+        let reset_early_group = builder
+            .component
+            .find_group(*group_name)
+            .unwrap_or_else(|| {
+                unreachable!(
+                    "called build_wrapper_group with {}, which is not a group",
+                    group_name
+                )
+            });
+        let early_reset_fsm =
+            builder.component.find_cell(*fsm_name).unwrap_or_else(|| {
+                unreachable!(
+                    "called build_wrapper_group with {}, which is not an fsm",
+                    fsm_name
+                )
+            });
+
+        let wrapper_group =
+            builder.add_group(format!("while_wrapper_{}", group_name));
+
+        structure!(
+            builder;
+            let one = constant(1, 1);
+            let time_0 = constant(0, fsm_width);
+        );
+
+        let port_parent = port.borrow().cell_parent();
+        let port_name = port.borrow().name;
+        let done_guard = (!guard!(port_parent[port_name]))
+            & guard!(early_reset_fsm["out"]).eq(guard!(time_0["out"]));
+
+        let assignments = build_assignments!(
+            builder;
+            reset_early_group["go"] = ? one["out"];
+            // cond_reg["in"] = time_guard_0 ? port_parent[port_name];
+            // cond_reg["write_en"] = time_guard_0 ? one["out"];
+            // cond_wire["in"] = time_guard_0 ? port_parent[port_name];
+            // cond_wire["in"] = not_0 ? cond_reg["out"];
```

Comment:
Are the commented lines intentional?

Reply:
I think we should delete those comments. I think they probably come from an earlier version of the pass that @paili0628 and I were discussing where we thought we might need to maintain the original value of the port throughout the while loop (@paili0628 can correct me if I'm wrong). But after thinking it over it turns out we don't need to do that.